src/pjm/parser.jl

```diff
@@ -0,0 +1,152 @@
+    TotalLMPDayAhead::DataFrame
+    CongestionPriceDayAhead::DataFrame
+    MarginalLossPriceDayAhead::DataFrame
+
+    function DayAheadHourlyLMP(start_date::String, start_hour_minute, end_date::String, end_hour_minute::String)
+        df = get_day_ahead_hourly_lmp(start_date, start_hour_minute, end_date, end_hour_minute)
+        df_dict, meta_df = parse_df_format(df, DayAheadHourlyLMP_values_keys, DayAheadHourlyLMP_meta_keys)
+        return new(
+            meta_df,
+            df_dict["system_energy_price_da"],
+            df_dict["total_lmp_da"],
+            df_dict["congestion_price_da"],
+            df_dict["marginal_loss_price_da"],
+        )
+    end
+end
+
+"""
+    get_day_ahead_hourly_lmp(start_date::String, end_date::String)::DataFrame
+
+Get the dataframe for the day ahead hourly lmp data from the PJM API between `start_date` and `end_date`.
+
+# Arguments
+- start_date::String: The start date of the data to be retrieved (format: "m/d/y").
+- end_date::String: The end date of the data to be retrieved (format: "m/d/y").
+- start_hour_minute::String: The start hour and minute of the data to be retrieved (format: "HH:MM").
+- end_hour_minute::String: The end hour and minute of the data to be retrieved (format: "HH:MM").
+
+# Returns
+- DataFrame: The dataframe for the day ahead hourly lmp data from the PJM API between `start_date` and `end_date`.
+"""
+function get_day_ahead_hourly_lmp(start_date::String, start_hour_minute, end_date::String, end_hour_minute)::DataFrame
+    access_key_dict = get_acess_key_headers()
+    url = get_url_day_ahead_hourly_lmp(start_date, start_hour_minute, end_date, end_hour_minute)
+    response = HTTP.get(url, headers=access_key_dict)
+
+    return CSV.read(IOBuffer(response.body), DataFrame)
+
+end
+
+"""
+ parse_df_format(df::DataFrame, value_keys::Vector{String}, meta_keys::Vector{String})::Tuple{Dict, DataFrame}
+
+parse the DataFrame into a Dict of DataFrames for each value key and a DataFrame for the meta keys.
+
+# Arguments
+- df::DataFrame: The DataFrame to be parsed.
+- value_keys::Vector{String}: The keys to be parsed into a Dict of DataFrames.
+- meta_keys::Vector{String}: The keys to be parsed into a DataFrame.
+
+# Returns
+- Tuple{Dict, DataFrame}: A Tuple containing the Dict of DataFrames for each value key and a DataFrame for the meta keys.
+"""
+function parse_df_format(df::DataFrame, value_keys::Vector{String}, meta_keys::Vector{String})::Tuple{Dict, DataFrame}
+
+    df.datetime_beginning_utc = DateTime.(df.datetime_beginning_utc, "m/d/y H:M:S p")
+    df.datetime_beginning_ept = DateTime.(df.datetime_beginning_ept, "m/d/y H:M:S p")
+
+    df_dict   = get_group_df(df, value_keys)
+    meta_df   = get_meta_dict(df, meta_keys)
+
+    return df_dict, meta_df
+end
+
+"""
+    get_group_df(df::DataFrame, value_keys::Vector{String})::Dict{String, DataFrame}
+
+Group the DataFrame by pnode_id and return a Dict of DataFrames for each value key.
+
```

Comment:
I am getting a bit confused with the transformations here. Could you show an example of the raw data and then the output transformed dataframes?

Reply:
I personally like the idea of having a function that returns a single dataframe (which may be parsed) and we can have an additional transformation helper function that makes those into multiple if the user wants it.